From 7458336bbfc4da894bdefc2621181000638d0afb Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Fri, 14 Apr 2023 19:46:32 +0530
Subject: [PATCH] cfg80211 : exported cfg80211_chandef_dfs_cac_time api

Driver needs this cfg80211_chandef_dfs_cac_time for calculating
the dfs_cac_time across the chandef

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 include/net/cfg80211.h | 12 ++++++++++++
 net/wireless/chan.c    |  1 +
 net/wireless/core.h    |  4 ----
 3 files changed, 13 insertions(+), 4 deletions(-)

--- a/include/net/cfg80211.h
+++ b/include/net/cfg80211.h
@@ -1071,18 +1071,6 @@ bool cfg80211_chandef_dfs_usable(struct
 				 const struct cfg80211_chan_def *chandef);
 
 /**
- * cfg80211_chandef_dfs_cac_time - get the DFS CAC time (in ms) for given
- *				   channel definition
- * @wiphy: the wiphy to validate against
- * @chandef: the channel definition to check
- *
- * Returns: DFS CAC time (in ms) which applies for this channel definition
- */
-unsigned int
-cfg80211_chandef_dfs_cac_time(struct wiphy *wiphy,
-			      const struct cfg80211_chan_def *chandef);
-
-/**
  * nl80211_send_chandef - sends the channel definition.
  * @msg: the msg to send channel definition
  * @chandef: the channel definition to check
@@ -1129,6 +1117,18 @@ ieee80211_chandef_rate_flags(struct cfg8
 }
 
 /**
+ * cfg80211_chandef_dfs_cac_time - get maximum dfs cac time traversing
+ * through the entire channel width
+ * @wiphy: the wiphy to validate against
+ * @chandef: the channel definition to check
+ * @is_bgcac: background cac state
+ * Returns: dfs cac time, or 0 is none found
+ */
+unsigned int cfg80211_chandef_dfs_cac_time(struct wiphy *wiphy,
+                                           const struct cfg80211_chan_def *chandef,
+                                           bool is_bgcac);
+
+/**
  * ieee80211_chandef_max_power - maximum transmission power for the chandef
  *
  * In some regulations, the transmit power may depend on the configured channel
--- a/net/wireless/chan.c
+++ b/net/wireless/chan.c
@@ -1205,6 +1205,7 @@ static bool cfg80211_secondary_chans_ok(
 
 	return true;
 }
+EXPORT_SYMBOL(cfg80211_chandef_dfs_cac_time);
 
 /* check if the operating channels are valid and supported */
 static bool cfg80211_edmg_usable(struct wiphy *wiphy, u8 edmg_channels,
